### PR TITLE
Implement SelectText

### PR DESCRIPTION
**Pull request by fanzhemeng**
This PR allows corresponding text in VS editor to be selected when user clicks on an Item in the tree view of SyntaxVisualizer.

### CODE REVIEW

**Comment by JoshVarty** [comment]
Can you add a description describing the overall changes you've made?

File: TypeScriptSyntaxVisualizer/MyControl.xaml.cs [hunk 1]
```diff
@@ -107,8 +111,18 @@ private void generateSyntaxTreeView(SyntaxNodeOrToken node, TreeViewItem item)
 
         private void selectText(TreeViewItem item)
         {
-            var node = (SyntaxNodeOrToken)item.DataContext;
-            //TODO
+            try {
```
Comment (JoshVarty) [inline TypeScriptSyntaxVisualizer/MyControl.xaml.cs]: Put the `{` on the newline. (This is just the standard in C# for some reason)

File: TypeScriptSyntaxVisualizer/TextViewCreationListener.cs [hunk 2]
```diff
@@ -26,11 +26,22 @@ private void Caret_PositionChanged(object sender, CaretPositionChangedEventArgs
 
                 //We roll the dice on an OOM exception... :(
                 string rawText = caret.ContainingTextViewLine.Snapshot.GetText();
+                string textbeforecaret = rawText.Substring(0, position);
+
+                int linenum = 0;
+                foreach (char c in textbeforecaret)
+                {
+                    if (c == '\r')
```
Comment (AmadeusW) [inline TypeScriptSyntaxVisualizer/TextViewCreationListener.cs]: Use either `\n` or `Environment.NewLine`.

File: TypeScriptSyntaxVisualizer/TextViewCreationListener.cs [hunk 3]
```diff
@@ -26,11 +26,22 @@ private void Caret_PositionChanged(object sender, CaretPositionChangedEventArgs
 
                 //We roll the dice on an OOM exception... :(
                 string rawText = caret.ContainingTextViewLine.Snapshot.GetText();
+                string textbeforecaret = rawText.Substring(0, position);
```
Comment (AmadeusW) [inline TypeScriptSyntaxVisualizer/TextViewCreationListener.cs]: use camelCase and verbose names for variables, for example `textBeforeCaret` and `lineNumber`

**Comment by fanzhemeng** [comment]
When testing I found there is an issue that after the text is selected, tree view of visualizer will change its focus to the item that corresponds to the end point of text selection. This needed to be solved by only allowing change with either `TextSelection` or `TreeViewItem` at one time, but not both.

**Comment by AmadeusW** [comment]
Looks good!
Perhaps rename `CaretPosChangeBecauseOfTextSlection` to `IgnoreNextCaretPositionChange`?

**Pull request merged**